WalletWasabi.Fluent/Controls/UriEntryBox.cs

```diff
@@ -0,0 +1,27 @@
+using Avalonia;
+using Avalonia.Controls;
+using WalletWasabi.Fluent.Helpers;
+
+namespace WalletWasabi.Fluent.Controls;
+
+public class UriEntryBox : TextBox
+{
+	protected override Type StyleKeyOverride => typeof(TextBox);
+
+	public UriEntryBox()
+	{
+		this.GetObservable(TextProperty).Subscribe(text =>
+		{
+			if (string.IsNullOrEmpty(text))
+			{
+				return;
+			}
+
+			var trimmedText = text.TotalTrim();
+			if (text != trimmedText)
+			{
+				Text = trimmedText;
+			}
+		});
```

Comment:
This is risky, you should modify the input during processing and not when it was already processed.
Hitting a char + space + char puts the caret in the wrong position.

![image](https://github.com/WalletWasabi/WalletWasabi/assets/16364053/9d402603-d61a-4422-aeb5-f67832cb4106)